apps/frontend/src/pages/projects/projects.tsx

```diff
@@ -1,30 +1,41 @@
 import { Loader, PageLayout } from "~/libs/components/components.js";
+import { EMPTY_ARRAY_LENGTH } from "~/libs/constants/constants.js";
 import { DataStatus } from "~/libs/enums/enums.js";
 import {
 	useAppDispatch,
 	useAppSelector,
+	useCallback,
 	useEffect,
 } from "~/libs/hooks/hooks.js";
 import { actions as projectActions } from "~/modules/projects/projects.js";
 
-import { ProjectCard } from "./components/components.js";
+import { ProjectCard, ProjectsSearch } from "./components/components.js";
 import styles from "./styles.module.css";
 
 const Projects = (): JSX.Element => {
 	const dispatch = useAppDispatch();
-
 	const { dataStatus, projects } = useAppSelector(({ projects }) => projects);
 
 	useEffect(() => {
 		void dispatch(projectActions.loadAll());
 	}, [dispatch]);
 
+	const handleSearchChange = useCallback(
+		(value: string) => {
+			void dispatch(projectActions.loadAll(value));
+		},
+		[dispatch],
+	);
+
 	const isLoading =
-		dataStatus === DataStatus.PENDING || dataStatus === DataStatus.IDLE;
+		dataStatus === DataStatus.IDLE ||
+		(dataStatus === DataStatus.PENDING &&
+			projects.length === EMPTY_ARRAY_LENGTH);
```

Comment:
```suggestion
		(dataStatus === DataStatus.PENDING &&
			hasProject);
```
decompose such variables